Sources/FoundationEssentials/Calendar/Calendar_Enumerate.swift

```diff
@@ -407,47 +407,43 @@ extension Calendar {
         }
         
         struct Iterator: Sendable, IteratorProtocol {
-            var calendar: Calendar
-            var previousResult: Date?
+            let calendar: Calendar
+            let start: Date
             let range: Range<Date>?
-            var components: DateComponents?
-            var wrappingComponents: Bool
+            let components: DateComponents
+            let wrappingComponents: Bool
+            var finished = false
+            var iteration = 1
             
             /// If `components` is `nil`, the result is an empty `Sequence`.
-            init(calendar: Calendar, start: Date, range: Range<Date>?, components: DateComponents?, wrappingComponents: Bool) {
+            init(calendar: Calendar, start: Date, range: Range<Date>?, components: DateComponents, wrappingComponents: Bool) {
+                self.start = start
                 self.calendar = calendar
-                previousResult = start
                 self.components = components
                 self.range = range
                 self.wrappingComponents = wrappingComponents
             }
             
             mutating func next() -> Element? {
-                guard let components else {
-                    // We have nothing to add
+                guard !finished else {
                     return nil
                 }
                 
-                guard let startAt = previousResult else {
-                    // We have finished
-                    return nil
-                }
-                                
-                let next = calendar.date(byAdding: components, to: startAt, wrappingComponents: wrappingComponents)
+                let scaled = components.scaled(by: iteration)
+                let next = calendar.date(byAdding: scaled, to: start, wrappingComponents: wrappingComponents)
                 guard let next else {
                     // We have finished
-                    previousResult = nil
+                    finished = false
                     return nil
                 }
```

Comment:
Shouldn't this be `finished = true`, since "We have finished"?

Reply:
Yes!